clients/src/main/java/org/apache/kafka/common/header/Headers.java

```diff
@@ -69,4 +69,10 @@ public interface Headers extends Iterable<Header> {
      */
     Header[] toArray();
 
+    /**
+     * Set Header to readonly.
+     */
+    default void setReadOnly() {
```

Comment:
Hmm, I don't think `setReadOnly` was ever intended to be public. The class `RecordHeaders` is in `internals`. We just use this method as a way to prevent modification once the headers have been passed to the producer.